model/distro/distro.go

```diff
@@ -119,6 +119,34 @@ func (d *Distro) IsParent(s *evergreen.Settings) bool {
 	return false
 }
 
+func (d *Distro) GetImageID() (string, error) {
+	var i interface{}
+	switch d.Provider {
+	case evergreen.ProviderNameEc2Auto:
+		i = (*d.ProviderSettings)["ami"]
+	case evergreen.ProviderNameEc2OnDemand:
+		i = (*d.ProviderSettings)["ami"]
+	case evergreen.ProviderNameEc2Spot:
+		i = (*d.ProviderSettings)["ami"]
+	case evergreen.ProviderNameDocker:
+		i = (*d.ProviderSettings)["image_url"]
+	case evergreen.ProviderNameDockerMock:
+		i = (*d.ProviderSettings)["image_url"]
+	case evergreen.ProviderNameGce:
+		i = (*d.ProviderSettings)["image_name"]
+	case evergreen.ProviderNameVsphere:
+		i = (*d.ProviderSettings)["template"]
+	default:
+		return "", nil
```

Comment:
should this be an error? or a thing we log?

Reply:
There are just some providers that do not have an image id, for example, the `Mock` and `Static`. I can check for those two (and leave the empty string as the image id), and if the Provider is unknown then throw an error. How does that sound?

Reply:
Done.